core/src/main/java/org/elasticsearch/index/mapper/core/ByteFieldMapper.java

```diff
@@ -174,8 +172,8 @@ public Query rangeQuery(Object lowerTerm, Object upperTerm, boolean includeLower
         }
 
         @Override
-        public Query fuzzyQuery(String value, Fuzziness fuzziness, int prefixLength, int maxExpansions, boolean transpositions) {
-            byte iValue = Byte.parseByte(value);
+        public Query fuzzyQuery(Object value, Fuzziness fuzziness, int prefixLength, int maxExpansions, boolean transpositions) {
+            byte iValue = Byte.parseByte(value.toString());
```

Comment:
All numeric mappers should use the same parse*Value that is used by `indexedValueForSearch` so that the behaviour is similar to termQuery().